### ISSUE TITLE
Reproduce the evaluation result

### ISSUE THREAD

**Issue by sunshantong** [hughw19/NOCS_CVPR2019#28]
Hi, I use pretrain weights to reproduce the evaluation results. The classification and regression evaluation results are as follows:

Classification results:
```
3D IoU at 25: 84.4
3D IoU at 50: 76.9
5 degree, 5cm: 9.8
5 degree, 100cm: 10.8
10 degree, 5cm: 24.0
10 degree, 10cm: 24.3
15 degree, 5cm: 34.7
15 degree, 10cm: 36.8
```

Regression results:
```
3D IoU at 25: 79.6
3D IoU at 50: 72.5
5 degree, 5cm: 7.7
5 degree, 100cm: 9.6
10 degree, 5cm: 21.7
10 degree, 10cm: 21.8
15 degree, 5cm: 32.9
15 degree, 10cm: 33.7
```
Most of the evaluation results are consistent with the results of the paper, except classification 32bins (_10 degree, 5cm: 25.2 10 degree, 10cm:25.8_ ) and regression ( _10 degree, 5cm:23.4  10 degree, 10cm:25.8_). It is 1.2 to 1.7 less than the paper. I am very confused and look forward to your reply. Thank you so much.

**Comment by Liuchongpei** [comment]
@sunshantong Hello, how did you set the epoch? According to the “Training Protocol” in the paper, it should be stage1:10, stage2:10+3, stage3:10+3+70, but the result is very poor. The code here is stage1:100, stage:100+30, stage3:100+30+270, and the result is better than in the paper.

**Comment by sunshantong** [comment]
@Liuchongpei Hi, I set the epoch like the code. But my training results are not as good as the paper. Is each item of your training results better than those in the paper? Thanks.

**Comment by Liuchongpei** [comment]
@sunshantong It seems so. The pre-training weight I used is mask_rcnn_coco.h5.
![c98123ba-b764-4d71-b7ca-24e6f99151cd_看图王](https://user-images.githubusercontent.com/34116796/88480657-c3c29580-cf89-11ea-9f94-b96ead647900.jpg)
 I changed the network from 32BIN to regression by changing the value pointed by the arrow (I'm not sure if this is correct), and the result is the same.
![2345截图20200726213435](https://user-images.githubusercontent.com/34116796/88480673-d5a43880-cf89-11ea-9d6c-8609bc3292a5.jpg)

**Comment by sunshantong** [comment]
@Liuchongpei Hi, Did you train for 400 epochs to get the 32BIN result? The results I got after 400 epochs training is not as good as in the paper. Your results look so much better. I am very confused. Do you know what this might be? BTW, you should set regression in _train.py_. 
   ```
 COORD_USE_BINS = False
    if COORD_USE_BINS:
         COORD_NUM_BINS = 32
    else:
        COORD_REGRESS_LOSS   = 'Soft_L1'
```

**Comment by Liuchongpei** [comment]
Yes, 400 epochs. I have no idea about your situation, but the epochs described in the paper are different from the code. And Thanks  for your reminder

**Comment by Liuchongpei** [comment]
@sunshantong Hi, after executing the command "python3 detect_eval.py --mode detect --ckpt_path = / logs / ckpt --draw", many GT pose visualization pictures (named "*bbox_gt.png") will be generated. It is obvious from these pictures that the GT poses of some objects are wrong. Have you encountered such a situation?
![2345截图20200813153632](https://user-images.githubusercontent.com/34116796/90108441-e2dc6800-dd7c-11ea-9f91-581881a4e393.jpg)
![2345截图20200813153709](https://user-images.githubusercontent.com/34116796/90108451-e7088580-dd7c-11ea-8f9d-9d495c89ff82.jpg)
![2345截图20200813153748](https://user-images.githubusercontent.com/34116796/90108458-ea037600-dd7c-11ea-8576-d22fac4dc544.jpg)

**Comment by hughw19** [comment]
Sorry for the inaccurate annotation. It is not meant to be wrong. The GT annotations are not done frame by frame. There are some propagation errors between two annotated frames. You may find some bad frames like what you showed.


Best,
He

**Comment by Liuchongpei** [comment]
@hughw19 Hello, thank you for your great work. However, inaccurate GT poses do not only occur in a few frames. At least in real_test_scene_1_0000_bbox_gt.png-real_test_scene_1_0172_bbox_gt.png and real_test_scene_1_0231_bbox_gt.png-real_test_scene_1_0383_bbox_gt.png, it is obvious that the pose annotation is not accurate and some errors are quite large. Other scenes have similar situations. I think such GT is difficult to use to evaluate the performance of the method. Can you provide more accurate annotation data? And if inaccurate GT is used in the process of generating *_coord.png, it should also have an negative impact on training.

**Comment by hughw19** [comment]
Hi Chongpei,

I currently don’t have the plan to improve the annotation accuracy. Using
real image for training is not a must. I still believe the annotation is
sufficient to examine the relative performance of algorithms. I would like
to improve the dataset later this year if I can get a working student.

Best,
He

**Comment by sunshantong** [comment]
Hi, @Liuchongpei I notice that you are also focusing on NOCS and 6-PACK. Can I talk to you in private? My email is shantongsun@foxmail.com  My WeChat is charplins
